packages/shared/input-with-requirements/TextInputWithFieldRequirements.tsx

```diff
@@ -0,0 +1,138 @@
+import * as React from 'react';
+import CheckCircleIcon from '@patternfly/react-icons/dist/esm/icons/check-circle-icon';
+import ExclamationTriangleIcon from '@patternfly/react-icons/dist/esm/icons/exclamation-triangle-icon';
+import InfoCircleIcon from '@patternfly/react-icons/dist/esm/icons/info-circle-icon';
+import cn from 'classnames';
+import { useController, Control, FieldValues } from 'react-hook-form';
+import {
+  FormGroup,
+  FormGroupProps,
+  InputGroup,
+  TextInput,
+  Popover,
+  PopoverPosition,
+  HelperText,
+  HelperTextItem,
+  Button,
+  TextInputProps,
+  Icon,
+} from '@patternfly/react-core';
+import useFieldRequirements from './useFieldRequirements';
+import './TextInputWithFieldRequirements.scss';
+
+export type TextInputWithFieldRequirementsProps = {
+  fieldRequirements: string[];
+  control: Control<FieldValues>;
+  formGroupProps: FormGroupProps;
+  textInputProps: TextInputProps & { ['data-test']: string };
+};
+
+export type ValidationIconProp = {
+  status: FormGroupProps['validated'];
+};
+
+export const ValidationIcon: React.FC<ValidationIconProp> = ({ status }) => {
+  const getStatusIcon = () => {
+    switch (status) {
```

Comment:
```suggestion
  const getStatusIcon = (currenStatus) => {
    switch (currentStatus) {
```
This way the function will be reusable and won't be recreated when the props change or update. If you want to keep it as it is use `useCallback`